src/app/services/app.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, Provider } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class AppService {
+  timer: Observable<any>;
+
+  constructor() {
+    this.timer = Observable.timer(1000, 1000).share();
```

Comment:
you don't need to `.share()` inner observables (you are using that one in `getCurrentTime` which is shared already).

Reply:
done